### PR TITLE
[Adjustments] Update line item enteprise fees

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Following #6858

Updates the enterprise fee adjustment association on line items to remove the need for all three polymorphic associations.

#### What should we test?

Enterprise fees on line items should work correctly.

#### Release notes

Updated associations on line item fees.

Changelog Category: Technical changes

### CODE REVIEW

File: app/services/order_adjustments_fetcher.rb [hunk 1]
```diff
@@ -29,14 +29,6 @@ def ship_total
     sum_adjustments "shipping"
   end
 
-  def line_item_adjustments(line_item)
```
Comment (mkllnk) [inline app/services/order_adjustments_fetcher.rb:32]: Great. This could have been squashed with the other commit.

**Review by mkllnk — APPROVED** [review]
The code is looking good. But it's difficult to tell if you covered all affected code. This needs some good testing again.

File: spec/models/spree/adjustment_spec.rb [hunk 2]
```diff
@@ -274,7 +274,7 @@ module Spree
         let(:order_cycle) { create(:simple_order_cycle, coordinator: coordinator, coordinator_fees: [enterprise_fee], distributors: [coordinator], variants: [variant]) }
         let(:line_item)   { create(:line_item, variant: variant) }
         let(:order)       { create(:order, line_items: [line_item], order_cycle: order_cycle, distributor: coordinator) }
-        let(:adjustment)  { order.adjustments(:reload).enterprise_fee.first }
+        let(:adjustment)  { order.all_adjustments(:reload).enterprise_fee.first }
```
Comment (sauloperez) [inline spec/models/spree/adjustment_spec.rb:277]: would you mind fixing the deprecation warning calling `#reload` instead?

File: app/models/spree/order.rb [hunk 3]
```diff
@@ -668,7 +668,7 @@ def shipping_tax
     end
 
     def enterprise_fee_tax
-      adjustments(:reload).enterprise_fee.sum(:included_tax)
+      all_adjustments(:reload).enterprise_fee.sum(:included_tax)
```
Comment (sauloperez) [inline app/models/spree/order.rb:671]: here as well

File: spec/models/spree/order_spec.rb [hunk 4]
```diff
@@ -521,6 +521,7 @@
   end
 
   describe "applying enterprise fees" do
+    let(:subject) { create(:order) }
```
Comment (sauloperez) [inline spec/models/spree/order_spec.rb:524]: This should rather be

```suggestion
    subject { create(:order) }
```

That's why it's called `subject` in the first place.

File: spec/models/spree/adjustment_spec.rb (hunk 2 above)
Comment (Matt-Yorkley) [reply spec/models/spree/adjustment_spec.rb:277]: https://github.com/openfoodfoundation/openfoodnetwork/pull/7132

File: spec/models/spree/line_item_spec.rb [hunk 5]
```diff
@@ -418,9 +418,7 @@ module Spree
         li = LineItem.new
 
         allow(li).to receive(:price) { 55.55 }
-        allow(li).to receive_message_chain(:order, :adjustments, :loaded?)
-        allow(li).to receive_message_chain(:order, :adjustments, :select)
-        allow(li).to receive_message_chain(:order, :adjustments, :where, :to_a, :sum) { 11.11 }
+        allow(li).to receive_message_chain(:adjustments, :enterprise_fee, :sum) { 11.11 }
```
[the remark this answers is not in the document]
Comment (Matt-Yorkley) [reply spec/models/spree/line_item_spec.rb:421]: Yeah it would be good to make some improvements here. This file is currently changing quite often...

**Review by sauloperez — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley,

This looks good: Compared the orders placed and respective fees on line items to previous orders:
(below **left**, before this PR; **right**, after this PR)

image

We can see the same enterprise fees are applied, and the ones which work on the line item level are now shown accordingly (in the upper section); values are correct.

Also, I checked the scenario mentioned on:
i) #7048 (related to subs) and
ii) #1629 (editing a previously placed order)

- Repeated the **test-case i)**, placing a subscription job, with an item out of stock:

image

We can see above, on the left side (admin view) that the enterprise fee is not charged anymore - It appears on the line items adjustments list, but its marked as £0 - the order total is correct :tada:

- Repeating **test case ii)**

Peek 2021-03-23 17-26

We can clearly see, that removing the line item, removes the enterprise fee as well :+1: It closes #1629.

Awesome, ready to go :mage_man:

**Pull request merged**